backend/src/routes/user.ts

```diff
@@ -112,4 +120,160 @@ router.post("/mentor", [validateMentor], async (req: Request, res: Response) =>
   }
 });
 
+/**
+ * This is a get route for a mentee. Note that the response is dependant on
+ * the person calling the method
+ *
+ * Mentee calling: gain all info about the mentee
+ *
+ * Mentor calling: mentee name, image, grade, about, career interests, topics of interest
+ */
+router.get("/mentee/:userId", [verifyAuthToken, upload], async (req: Request, res: Response) => {
+  const userId = req.params.userId;
+  if (!mongoose.Types.ObjectId.isValid(userId)) {
+    return res
+      .status(ServiceError.INVALID_MONGO_ID.status)
+      .send(ServiceError.INVALID_MONGO_ID.message);
+  }
+  const role = req.body.role;
+
+  if (role === "mentee") {
+    try {
+      const mentee = await Mentee.findById(userId);
+      if (!mentee) {
+        throw ServiceError.MENTEE_WAS_NOT_FOUND;
+      }
+      return res.status(201).send({
+        message: `Here is mentee ${mentee.name}`,
+        mentee,
+      });
+    } catch (e) {
+      console.log(e);
+      if (e instanceof ServiceError) {
+        return res.status(e.status).send(e.displayMessage(true));
+      }
+      return res
+        .status(InternalError.ERROR_GETTING_MENTEE.status)
+        .send(InternalError.ERROR_GETTING_MENTEE.displayMessage(true));
+    }
+  }
+
+  if (role === "mentor") {
+    try {
+      const mentee = await Mentee.findById(userId);
+      if (!mentee) {
+        throw ServiceError.MENTEE_WAS_NOT_FOUND;
+      }
+      const { name, imageId, about, grade, topicsOfInterest, careerInterests } = mentee;
+      return res.status(201).send({
+        message: `Here is mentee ${mentee.name}`,
+        mentee: {
+          name,
+          imageId,
+          about,
+          grade,
+          topicsOfInterest,
+          careerInterests,
+        },
+      });
+    } catch (e) {
+      console.log(e);
+      if (e instanceof ServiceError) {
+        return res.status(e.status).send(e.displayMessage(true));
+      }
+      return res
+        .status(InternalError.ERROR_GETTING_MENTEE.status)
+        .send(InternalError.ERROR_GETTING_MENTEE.displayMessage(true));
+    }
+  }
+
+  return res.status(405).send("Mentee or Mentor not accessing data");
+});
+
+/**
+ * This is a get route for a mentor. Note that the response is dependant on the person
+ * calling the method
+ *
+ * Mentee calling: mentor name, image, major, minor, college, career, graduation year, calendlyLink,
+ * why were you paired?, about, topics of experties
+ *
+ * Mentor calling: gain all info about the mentor
+ */
+router.get("/mentor/:userId", [verifyAuthToken, upload], async (req: Request, res: Response) => {
```

Comment:
Need to add 
- menteeId (if request is sent by the mentor's mentee)
- menteeIds (if requested by the mentor themselves)

Reply:
Fixed!